EIPS/eip-7862.md

```diff
@@ -0,0 +1,149 @@
+---
+eip: 7862
+title: Delayed Execution Layer State Root
+description: Delays the ExecutionPayload's state root reference by one block.
+author: Charlie Noyes <charlie@paradigm.xyz>, Dan Robinson <dan@paradigm.xyz>, Justin Drake <justin@ethereum.org>, Toni Wahrstätter (@nerolation)
+discussions-to: https://ethereum-magicians.org/t/eip-7862-delayed-execution-layer-state-root/22559
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-12-23
+---
+
+## Abstract
+
+This EIP proposes that, for each block `n`, the `state_root` in block `n`'s `ExecutionPayload` references the **post-state of block `n-1`**. As a result, block `n` no longer needs to include or verify the newly computed state root for block `n`’s own transactions. This removes the expensive state root computation from the block-propagation critical path, reducing end-to-end latency for block production and potentially increasing throughput.
+
+## Motivation
+
+Currently, every Ethereum block includes two `state_root`. 
+
+The consensus layer (CL) `state_root` contained within the `BeaconBlock` tracks CL state (e.g., validator entries and exits).
+
+The execution layer (EL) `state_root` contained within the `ExecutionPayload` tracks the effect of transaction execution (e.g., account balances, code, etc.). 
+
+The EL `state_root` must be computed by block builders (and verified by relays, validators, and light clients) in near-real-time. This computation accounts for a large fraction of block production and verification time, especially in an MEV-Boost environment. This overhead is also a challenge for efforts to enable real-time ZK proving of the chain.
```

Comment:
I'd love to see some numbers here, what is this large fraction? If splitting the time between receiving a block, executing the txs, and calculating the root, what would these numbers be?

Reply:
for purposes of draft EIP can we add a <-- TODO benchmark --> here?